packages/components/src/components/domainproperties/DomainRow.tsx

```diff
@@ -441,14 +442,15 @@ export class DomainRow extends React.PureComponent<IDomainRowProps, IDomainRowSt
                     isFieldFullyLocked(field.lockType) ||
                     isFieldPartiallyLocked(field.lockType) ||
                     isPrimaryKeyFieldLocked(field.lockType)
-                ) && (
-                    <DeleteIcon
-                        id={createFormInputId(DOMAIN_FIELD_DELETE, domainIndex, index)}
-                        title="Remove field"
-                        iconCls="domain-field-delete-icon"
-                        onDelete={this.onDelete}
-                    />
-                )}
+                ) &&
+                    !domainFormDisplayOptions.hideDeleteIcon && (
```

Comment:
This check now has 3 booleans OR'd together which is negated and then another boolean negation. I think it is time to pull this into a helper function, something like:
```
showDeleteIcon(field): boolean {
    return !isFieldFullyLocked(field.lockType) && !isFieldPartiallyLocked(field.lockType) && !isPrimaryKeyFieldLocked(field.lockType) && !domainFormDisplayOptions.hideDeleteIcon;
}
```